enterprise_access/apps/subsidy_access_policy/models.py

```diff
@@ -260,14 +260,29 @@ class Meta:
 
     def can_redeem(self, learner_id, content_key):
         learner_transaction_count = subsidy_client.transactions_for_learner(
-            subsidy_uuid = self.subsidy_uuid,
+            subsidy_uuid=self.subsidy_uuid,
             learner_id = learner_id,
         )
         if learner_transaction_count < self.per_learner_enrollment_limit:
             return super().can_redeem(learner_id, content_key)
 
         return False
 
+    def credit_available(self, learner_id=None):
+        if self.remaining_credit(learner_id) > 0:
+            return True
+        return False
+
+    def remaining_credit(self, learner_id=None):
+        """
+        Returns the remaining redeemable credit for the policy.
+        """
+        learner_transaction_count = subsidy_client.transactions_for_learner(
```

Comment:
[curious] Is the intent for `transactions_for_learner` only to return transactions in a "committed" (i.e., fulfilled) state and without any reversals that would negate the transaction?

Reply:
For example, say a learner enrolls in a course and then unenrolls with the 14 refund day window so a transaction reversal is created to add funds back to the ledger balance. In that case, would there be 2 transactions for the learner that negate each other?

Reply:
I think existing Transactions in any state besides "committed" should count towards the consumed credit for a policy because they represent in-process attempts to redeem content, and ignoring other states could result in a race condition of over-allocation of redemptions.  That said, policy-level locking should prevent this race condition in the first place.

Reply:
Regarding reversals: I can't speak to those, it sounds like a product question for standup.

Reply:
@adamstankiewicz That's a good point. I think this logic should be handled at `enterprise-subsidy` side.

Reply:
> I think existing Transactions in any state besides "committed" should count towards the consumed credit for a policy because they represent in-process attempts to redeem content

@pwnage101 Agreed, I could see Transactions in a created/pending state also count towards the consumed credit, but what about a transaction that we've failed to fulfill (e.g., learner not actually enrolled)? Looking at [`TransactionStateChoices`](https://github.com/openedx/openedx-ledger/blob/35e2b1ce22740f649ec3385adfa352e94797104d/openedx_ledger/models.py#L25-L38), I wonder if there should be an error state for transactions?

> Regarding reversals: I can't speak to those, it sounds like a product question for standup.

The reversals suggestion comes from a recent bug in ecommerce with existing learner credit where we didn't consider refunds when querying for learners' prior orders, so they weren't able to re-enroll because ecommerce thought they had already placed an order. I'll bring this up in standup then (likely a Milestone 2 thing anyways).